scripts/irods/test/test_iput.py

```diff
@@ -886,6 +887,50 @@ def test_large_put(self):
         self.object_name = 'test_large_put'
         self.run_tests(self.object_name, 5120000)
 
+    def test_iput_with_updated_checksum_read_buffer_size_in_bytes__issue_7947(self):
+        self.object_name = 'test_iput_with_updated_checksum_read_buffer_size_in_bytes__issue_7947'
+        irodsctl = IrodsController()
+        server_config_filename = paths.server_config_path()
+
+        # load server_config.json to inject a new rule base
+        with open(server_config_filename) as f:
+            svr_cfg = json.load(f)
+
+        # update the checksum_read_buffer_size_in_bytes to 10 MiB
+        svr_cfg['advanced_settings']['checksum_read_buffer_size_in_bytes'] = 10*1024*1024
+
+        # dump to a string to repave the existing server_config.json
+        new_server_config = json.dumps(svr_cfg, sort_keys=True, indent=4, separators=(',', ': '))
+
+        # create a 100 MiB file
+        file_size = 100*1024*1024
+        local_file = os.path.join(self.admin.local_session_dir, self.object_name)
+        lib.make_file(local_file, file_size)
+
+        # put the file using the original server configuration and read the calculated checksum
+        self.admin.assert_icommand(['iput', '-R', self.resource, '-k', local_file, self.object_name])
+        tmp_checksum = self.get_checksum(self.object_name)
+
+        # clean up this file
+        self.admin.assert_icommand(['irm', '-f', self.object_name])
+
+        with lib.file_backed_up(server_config_filename):
+            # repave the existing server_config.json
+            with open(server_config_filename, 'w') as f:
+                f.write(new_server_config)
+
+            # Bounce server to apply setting
+            irodsctl.restart()
+
+            # put the file again with new server configuration and compare checksum
+            self.admin.assert_icommand(['iput', '-R', self.resource, '-k', local_file, self.object_name])
+            self.assertEqual(self.get_checksum(self.object_name), tmp_checksum)
+
+        # self.object_name deleted in teardown
+
+        # Bounce server to get back original settings
+        irodsctl.restart()
```

Comment:
The restart instructions can be removed since every agent reloads server_config.json after a fork.

Reply:
done